server/routes/induction/update/previousWorkExperienceTypesUpdateController.ts

```diff
@@ -0,0 +1,251 @@
+import createError from 'http-errors'
+import { NextFunction, Request, RequestHandler, Response } from 'express'
+import type { InductionDto, PreviousWorkExperienceDto } from 'inductionDto'
+import type { PreviousWorkExperienceTypesForm } from 'inductionForms'
+import type { PrisonerSummary } from 'viewModels'
+import logger from '../../../../logger'
+import PreviousWorkExperienceTypesController from '../common/previousWorkExperienceTypesController'
+import { InductionService } from '../../../services'
+import validatePreviousWorkExperienceTypesForm from './previousWorkExperienceTypesFormValidator'
+import TypeOfWorkExperienceValue from '../../../enums/typeOfWorkExperienceValue'
+import toCreateOrUpdateInductionDto from '../../../data/mappers/createOrUpdateInductionDtoMapper'
+import previousWorkExperienceTypeScreenOrderComparator from '../previousWorkExperienceTypeScreenOrderComparator'
+
+export default class PreviousWorkExperienceTypesUpdateController extends PreviousWorkExperienceTypesController {
+  constructor(private readonly inductionService: InductionService) {
+    super()
+  }
+
+  getBackLinkUrl(req: Request): string {
+    const { prisonNumber } = req.params
+    return `/plan/${prisonNumber}/view/work-and-interests`
+  }
+
+  getBackLinkAriaText(_req: Request): string {
+    return 'Back to <TODO - check what CIAG UI does here>'
+  }
+
+  submitPreviousWorkExperienceTypesForm: RequestHandler = async (
+    req: Request,
+    res: Response,
+    next: NextFunction,
+  ): Promise<void> => {
+    const { prisonNumber } = req.params
+    const { prisonerSummary, inductionDto } = req.session
+
+    req.session.previousWorkExperienceTypesForm = { ...req.body }
+    if (!req.session.previousWorkExperienceTypesForm.typeOfWorkExperience) {
+      req.session.previousWorkExperienceTypesForm.typeOfWorkExperience = []
+    }
+    if (!Array.isArray(req.session.previousWorkExperienceTypesForm.typeOfWorkExperience)) {
+      req.session.previousWorkExperienceTypesForm.typeOfWorkExperience = [
+        req.session.previousWorkExperienceTypesForm.typeOfWorkExperience,
+      ]
+    }
+    const { previousWorkExperienceTypesForm } = req.session
+
+    const errors = validatePreviousWorkExperienceTypesForm(previousWorkExperienceTypesForm, prisonerSummary)
+    if (errors.length > 0) {
+      req.flash('errors', errors)
+      return res.redirect(`/prisoners/${prisonNumber}/induction/previous-work-experience`)
+    }
+
+    if (
+      noChangesToPreviousWorkExperienceSubmitted(inductionDto, previousWorkExperienceTypesForm) ||
+      onlyRemovalsWereMadeToPreviousWorkExperienceSubmitted(inductionDto, previousWorkExperienceTypesForm)
+    ) {
+      if (onlyRemovalsWereMadeToPreviousWorkExperienceSubmitted(inductionDto, previousWorkExperienceTypesForm)) {
+        logger.debug(
+          'Previous Work Experiences changes are only removed types so Induction can be updated without starting a flow',
```

Comment:
Minor `...without asking for work experience details` might more sense to the new reader?

Reply:
Yes, 💯 , will do that now 👍

Reply:
Done 👍